code/socialDistribution/forms.py

```diff
@@ -83,4 +133,27 @@ def clean_visibility(self):
         if data in [visibility[0] for visibility in LocalPost.Visibility.choices]:
             return data
         else:
-            raise ValidationError('Invalid visibility')
\ No newline at end of file
+            raise ValidationError('Invalid visibility')
+
+    def get_content_and_type(self):
+        """
+            Gets the content type and the content of the submitted data
+        """
+        post_type = self.cleaned_data.get('post_type')
+        image, image_binary, content_type = self.cleaned_data.get('image')
+        
+        # Upload image to media folder and base64-
+        # encode the data to store in database
+        if post_type == PostForm.IMAGE and image:
+            Image.objects.create(
+                caption=self.cleaned_data.get('description'),
+                image=image
+            )
+
+            content = base64.b64encode(image_binary)
+
+        else:
+            content_type = LocalPost.ContentType.PLAIN
+            content = base64.b64encode(self.cleaned_data.get('content_text'))
```

Comment:
Do we need to base64 encode text data? (just wondering)

Reply:
Hmm yes, you're right 🤔 I think we can just remove the b64encoding and just do an encode('utf-8') for storing since textual content won't be sent over as binary in any case. I don't know why I thought it would lol.  Thanks a lot for noticing that! ♥ This makes things easier.